File: CPP/13300_방배정.cpp
#include <iostream>
using namespace std;

int girl[6], boy[6];

int main() {
    int N, K, S, H, result = 0;
    cin >> N >> K;
    for (int i = 0; i < N; ++i) {
        cin >> S >> H;
        if (S == 0) {
            girl[H - 1] += 1;
            if (girl[H - 1] == K) {
                result += 1;
                girl[H - 1] = 0;
            }
        } else if (S == 1) {
            boy[H - 1] += 1;
            if (boy[H - 1] == K) {
                result += 1;
                boy[H -1] = 0;
            }
        }
    }
    for (int j = 0; j < 6; ++j) {
        if (girl[j] != 0) result += 1;
        if (boy[j] != 0) result += 1;
    }
    cout << result << endl;
    return 0;
}

File: Python/11:2 배열/13300_방배정.py
N, K = map(int, input().split())    # 학생수, 방 인원수

girl = [0] * 6
boy = [0] * 6

for _ in range(N):
    S, H = map(int, input().split())
    if S == 0:
        girl[H-1] += 1
    elif S == 1:
        boy[H-1] += 1

result = 0
for i in range(6):
    if girl[i] % K == 0:
        result += girl[i] // K
    elif girl[i] % K != 0:
        result += girl[i] // K + 1
    if boy[i] % K == 0:
        result += boy[i] // K
    elif boy[i] % K != 0:
        result += boy[i] // K + 1
print(result)